controllers/glanceapi_controller.go

```diff
@@ -799,3 +742,93 @@ func (r *GlanceAPIReconciler) createHashOfInputHashes(
 	}
 	return hash, changed, nil
 }
+
+// ensureKeystoneEndpoints -  create or update keystone endpoints
+func (r *GlanceAPIReconciler) ensureKeystoneEndpoints(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *glancev1.GlanceAPI,
+	serviceLabels map[string]string,
+) (ctrl.Result, error) {
+
+	var ctrlResult ctrl.Result
+	var err error
+
+	// If the parent controller didn't set the annotation, the current glanceAPIs
+	// shouldn't register the endpoints in keystone
+	if len(instance.ObjectMeta.Annotations) == 0 ||
+		instance.ObjectMeta.Annotations[glance.KeystoneEndpoint] != "true" {
+		// Mark the KeystoneEndpointReadyCondition as True because there's nothing
+		// to do here
+		instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.KeystoneEndpointReadyCondition))
+		instance.Status.Conditions.MarkTrue(
+			condition.KeystoneEndpointReadyCondition, condition.ReadyMessage)
```

Comment:
I'm confused here.  Why do we `Mirror()` if we're just overriding it with with `MarkTrue()` on the next line?

Reply:
Right, good catch, thanks. We don't need to `Mirror` the the condition as we're explicitly setting it to `True` (as per the comment in L762). I'm going to remove the `Mirror()` call because we don't need it here.